src/Mappers/BaseMapper.php

```diff
@@ -166,54 +164,51 @@ protected static function parseMoneyAttribute(?float $value): ?Money
         return Util::parseMoney($value, new Currency('EUR'));
     }
 
-    protected static function parseObjectAttribute(string $className, $object, \DOMElement $element, string $fieldTagName, array $attributes = null)
+    /** @var SomeClassWithMethodsetCode $object2 */
+    protected static function parseObjectAttribute(string $objectClass, $object, \DOMElement $element, string $fieldTagName, array $attributes = [])
     {
-        if ($className == "DimensionGroupDimension" || $className == "UnknownDimension") {
-            if ($className == "DimensionGroupDimension") {
+        if ($objectClass == "DimensionGroupDimension" || $objectClass == "UnknownDimension") {
```

Comment:
Should these strings be updated to class references (`::class`)?

Reply:
No, they cannot, as they are non existent stubs that tell this function to determine the real class from an argument contained in the XML element or sibling XML element.

`parseObjectAttribute` is usually used when the Twinfield entity contained in the XML is static (such as the <user> element which will always contain a user (or nothing)). In this case `::class` is used when calling `parseObjectAttribute`

In DimensionGroupDimension, CustomerFinancials, FixedAssetTransactionLine and SupplierFinancials there are fields that can contain multiple or any of the 7 dimensionTypes. In these cases` $className == "DimensionGroupDimension"` or `$className == "UnknownDimension"` is used to tell the function to change `$objectClass` to the real class name in the following `switch`

Reply:
Moved this part of the parseObjectAttribute function to a different function, seems like a cleaner solution